sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1393,6 +1393,10 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
           }
           str.charAt(0)
         }.getOrElse('\\')
+        if (Set('%', '_').contains(escapeChar)) {
```

Comment:
nit: don't use `Set` for just 2 elements. We can just do `a = % || a = _`

Reply:
Done.